src/Detector/DetectorLibrary/Exceptions/UnsupportedPlatformException.cs

```diff
@@ -0,0 +1,18 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+namespace Microsoft.Oryx.Detector.Exceptions
+{
+    /// <summary>
+    /// Supplied platform is not supported.
+    /// </summary>
+    public class UnsupportedPlatformException : InvalidUsageException
```

Comment:
This exception should not be required. We just have a bunch of detectors which will detect and give us back info.